### ISSUE TITLE
GCM message parameters names incorrect

### ISSUE THREAD

**Issue by marcospaulomc** [Ph3nol/NotificationPusher#81]
File: sly\notification-pusher\src\Sly\NotificationPusher\Adapter\Gcm.php
Line: 117

Parameters accepted at method getDefinedParameters are 'collapse_key', 'delay_while_idle', 'time_to_live', 'restricted_package_name', 'dry_run', but at getServiceMessageFromOrigin they are wrong

```
public function getServiceMessageFromOrigin(array $tokens, BaseOptionedModel $message)
    {
        $data            = $message->getOptions();
        $data['message'] = $message->getText();

        $serviceMessage = new ServiceMessage();
        $serviceMessage->setRegistrationIds($tokens);
        $serviceMessage->setData($data);
        $serviceMessage->setCollapseKey($this->getParameter('collapseKey'));
        $serviceMessage->setRestrictedPackageName($this->getParameter('restrictedPackageName'));
        $serviceMessage->setDelayWhileIdle($this->getParameter('delayWhileIdle', false));
        $serviceMessage->setTimeToLive($this->getParameter('ttl', 600));
        $serviceMessage->setDryRun($this->getParameter('dryRun', false));

        return $serviceMessage;
    }
```

```
public function getDefinedParameters()
    {
        return array(
            'collapse_key',
            'delay_while_idle',
            'time_to_live',
            'restricted_package_name',
            'dry_run'
        );
    }
```

**Comment by Knallcharge** [comment]
ran into this today, too, was wondering if it was just me :)

**Comment by pavankumarkatakam** [comment]
Yeah, it is problematic.

**Comment by pavankumarkatakam** [comment]
Has somebody a solution or fix for this ?

**Comment by pavankumarkatakam** [comment]
Is this fixed?

**Comment by fieteboerner** [comment]
@Ph3nol it would be very nice if you merge this and create a new release, cause I really need this function.  Apart from that: really helpful package.

Thanks in advance!

**Comment by seyfer** [comment]
@fieteboerner I decided to merge another pull request because it fixes all parameter names.
https://github.com/Ph3nol/NotificationPusher/issues/144

**Comment by seyfer** [comment]
Closed by #127
